ext/intl/msgformat/msgformat_helpers.cpp

```diff
@@ -439,31 +439,30 @@ U_CFUNC void umsg_format_helper(MessageFormatter_object *mfo,
 			switch (argType) {
 			case Formattable::kString:
 				{
-					zend_string *str, *tmp_str;
-
 	string_arg:
 					/* This implicitly converts objects
 					 * Note that our vectors will leak if object conversion fails
 					 * and PHP ends up with a fatal error and calls longjmp
 					 * as a result of that.
 					 */
-					str = zval_get_tmp_string(elem, &tmp_str);
+					if (!try_convert_to_string(elem)) {
```

Comment:
I believe this will implicitly modify the array. Can we skip the entire iteration when the parameter is unsed?